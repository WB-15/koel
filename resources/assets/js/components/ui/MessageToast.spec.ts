import { expect, it, vi } from 'vitest'
import UnitTestCase from '@/__tests__/UnitTestCase'
import MessageToast from './MessageToast.vue'
import { fireEvent } from '@testing-library/vue'

new class extends UnitTestCase {
  private renderComponent () {
    return this.render(MessageToast, {
      props: {
        message: {
          id: 101,
          type: 'success',
          message: 'Everything is fine',
          timeout: 5
        }
      }
    })
  }

  protected test () {
    it('renders', () => expect(this.renderComponent().html()).toMatchSnapshot())

    it('dismisses upon click', async () => {
      const { emitted, getByTitle } = this.renderComponent()
      await fireEvent.click(getByTitle('Click to dismiss'))

      expect(emitted().dismiss).toBeTruthy()
    })

    it('dismisses upon timeout', async () => {
      vi.useFakeTimers()

      const { emitted } = this.renderComponent()
      vi.advanceTimersByTime(5000)
      expect(emitted().dismiss).toBeTruthy()

      vi.useRealTimers()
    })
  }
}
