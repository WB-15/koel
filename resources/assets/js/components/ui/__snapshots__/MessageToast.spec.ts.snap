// Vitest Snapshot v1

exports[`renders 1`] = `
<div class="message success" title="Click to dismiss" data-v-f05fd058="">
  <aside data-v-f05fd058=""><br data-testid="icon" icon="[object Object]" class="icon" data-v-f05fd058=""><br data-testid="icon" icon="[object Object]" class="icon-dismiss" data-v-f05fd058=""></aside>
  <main data-v-f05fd058=""></main>
</div>
`;
